### ISSUE TITLE
Use singular_name for block type name by default

### ISSUE THREAD

**Issue by chillu** [dnadesign/silverstripe-elemental#619]
`BaseElement->getType()` returns `_t(__CLASS__ . '.BlockType', 'Block')`. That type is used in the UI, e.g. type dropdowns. For one, it's a fairly unusual name for a UI label ("type" typically refers to schema-related stuff). But ignoring that, it's also duplicating core functionality: We already have `DataObject->singular_name` for this purpose, which also hooks into automatic pluralisation. And that supports `_t()` through `i18n_singular_name()`. 

I'm suggesting that we should return `i18n_singular_name()` in `getType()`, which then defaults to the class name - rather than calling every new block type "block" by default. This wouldn't break existing implementations of `getType()`, but it would make it less confusing to seasoned SilverStripers like me who just assume that it uses `singular_name` like virtually everything else in the CMS UI :D

**Comment by robbieaverill** [comment]
Yeah we did this in order to be able to follow design patterns more closely, e.g. "Content" and "Content Block" are different (getType() is the former and i18n_singular is the latter) and are useful in or out of the context of viewing an elemental editor.

"Content" is used when referring to blocks in an elemental editor, because you already know it's a block, but "Content Block" would be used outside of this e.g. in ORM relationships or something because you don't necessarily know what a "Content" is when it's related to a page or a member or something.

Related conversations: https://github.com/dnadesign/silverstripe-elemental/issues/503 https://github.com/dnadesign/silverstripe-elemental/issues/249

Without getType() our add new functionality would contain a bunch of "Block" suffixes which we don't want. We could strip "Block" off the end, but it is variable so not always guaranteed to be "Block".

**Comment by ScopeyNZ** [comment]
I agree with @chillu that is seemed weird to me as well (hence #503). @robbieaverill has a good point too though. 

> but it is variable so not always guaranteed to be "Block"

I feel this way too, but I wonder if we could just make a decision and say they're all called blocks from now. Then we can just have a `{type} block` translation and then using `singular_name` to handle "type".

**Comment by robbieaverill** [comment]
I'd be OK with that. In favour of not over engineering a solution, perhaps we should support stripping of "Block" from the RHS but nothing else. non "XYZ Block" types would have a full i18n description in the dialog but it'd be edge case so OK. So the move for this would be to deprecate getType() in favour of i18n singular and use that with Block stripped off in the place of getType() etc

**Comment by GuySartorelli** [comment]
`i18n_singular_name` is called from `getType()` currently.
